libs/emblem/src/components/market-emblem.tsx

```diff
@@ -0,0 +1,57 @@
+import { URL_BASE } from '../config';
+import { EmblemBase } from './emblem-base';
+import { useMarketInfo } from './hooks/use-market-info';
+import { getVegaChain } from './lib/get-chain';
+
+export type EmblemByMarketProps = {
+  market: string;
+  vegaChain?: string;
+  showBase?: boolean;
+  showQuote?: boolean;
+  contract?: never;
```

Comment:
To me there doesn't seem to be any reason to include contract and asset if they are typed as `never`. You should get a warning if you provide the prop and its not declared in props

Reply:
I've used partial destructuring in a follow up. My reason for doing it this way was to pass `props` down to the underlying `Emblem` component directly, so they can also be controlled.

But now I'm not sure if that's actually a good idea or not. It feels a bit spaghetti like. I vote I just remove that ability. How does that feel?